SwiftIntro.playground/Contents.swift

```diff
@@ -284,72 +372,81 @@ var arrayOfItems2: [Item?] = [
     Item(name: "Dzinsai", category: .jeans)
 ]
 
-//var arrayOfShoes: [Item]
-//arrayOfShoes.forEach {
-//    print($0.name)
-//}
+var arrayOfShoes: [Item] = arrayOfItems2.compactMap({ $0 }).filter { $0.category == .shoes }
+
+arrayOfShoes.forEach {
+    print($0.name)
+}
 
 //MARK: DELEGATES
 //Rewrite it using delegate pattern
 
+protocol NetworkDelegate: AnyObject {
```

Comment:
Not sure ar cia geras sprendimas del retain cycle? Ir ar apskritai jis yra? Nes 457 eiluteje priskyrus nil kitaip nepasileidzia deinit() metodai.

Reply:
Actually, your solution is a combination of closures and delegate pattern. It should be one or another.
Personally, I chose delegate pattern if there are a lot of functions to call, and closures if there are only few of them.
In this case you should have functions instead of closures in your protocol, because, you already have reference (delegate) to your protocol in `NetworkService`. + You don't have to handle weak selfs for each closure